src/mo_frontend/typing.ml

```diff
@@ -2921,12 +2928,12 @@ and infer_dec_valdecs env dec : Scope.t =
 
 (* Programs *)
 
-let infer_prog scope pkg_opt async_cap prog : (T.typ * Scope.t) Diag.result =
+let infer_prog ?(viper_mode=false) scope pkg_opt async_cap prog : (T.typ * Scope.t) Diag.result =
```

Comment:
It would be nice (but not essential) to avoid the `?(viper_mode=false)` argument (here and below) by passing an extra field in scope instead.  
`